### PR TITLE
Fix storage endianness

### PR DESCRIPTION
**Pull request by dan-starkware**
## Pull Request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
Serialization endianness is not deterministic. This leads to wrong DB keys comparison.

Issue Number: N/A

## What is the new behavior?
Fixed big-endiannes. 

-

## Does this introduce a breaking change?

- [x] Yes
- [ ] No
New DB schema may not be backward compatible.

## Other information

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/568)

### CODE REVIEW

File: crates/papyrus_storage/src/serializers_test.rs [hunk 1]
```diff
@@ -44,6 +45,28 @@ pub(crate) use create_storage_serde_test;
...
-create_storage_serde_test!(StarkHash);
+create_storage_serde_test!(bool);
 create_storage_serde_test!(ContractAddress);
+create_storage_serde_test!(StarkHash);
 create_storage_serde_test!(StorageKey);
+create_storage_serde_test!(u8);
+create_storage_serde_test!(usize);
+
+#[test]
+fn block_number_endianness() {
```
Comment (dan-starkware) [inline crates/papyrus_storage/src/serializers_test.rs:56]: Fails before this PR changes.

File: crates/papyrus_storage/src/state/state_test.rs [hunk 2]
```diff
@@ -363,3 +363,50 @@ fn revert_state() {
         StarkFelt::from(0)
     );
 }
+
+#[test]
+fn get_nonce_key_serialization() {
```
Comment (dan-starkware) [inline crates/papyrus_storage/src/state/state_test.rs:368]: Fails before this PR changes.

**Review by spapinistarkware — APPROVED** [review]
:lgtm:

> Reviewed 7 of 7 files at r1, 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/568#-:-NROBeT3EFT1zUpMv_Gq:bnfp4nl)* status: 7 of 12 files reviewed, 2 unresolved discussions

**Review by spapinistarkware — COMMENTED** [review]
> Reviewed 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/568)* status: 8 of 12 files reviewed, 2 unresolved discussions

**Review by yair-starkware — DISMISSED** [review]
> Reviewed 6 of 7 files at r1, 2 of 2 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/568)* status: all files reviewed, 3 unresolved discussions (waiting on @dan-starkware and @spapinistarkware)

___
*[`crates/papyrus_storage/src/serializers.rs` line 117 at r3](https://reviewable.io/reviews/starkware-libs/papyrus/568#-NRSZg6d4ts6eQu-nrBY:-NRSZg6eDTZVmKPeqU6V:bkitlr) ([raw file](https://github.com/starkware-libs/papyrus/blob/1c3e14b12b264dabd4fb4fb845a1ab48ebd8c1e8/crates/papyrus_storage/src/serializers.rs#L117)):*
> ```XML
>         L1Handler = 2,
>     }
>     pub struct EthAddress(pub H160);
> ```

Consider implementing the trait for EthAddress instead of specifying what is H160

**Review by spapinistarkware — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/568#-:-NRSrjxNAPEjGkxGXPnQ:bnfp4nl)* status: all files reviewed, 3 unresolved discussions (waiting on @dan-starkware and @yair-starkware)

**Review by spapinistarkware — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/568)* status: all files reviewed, 1 unresolved discussion (waiting on @dan-starkware)

**Pull request merged**